src/Symfony/Component/Workflow/Metadata/MetadataStoreInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Metadata;
+
+use Symfony\Component\Workflow\Transition;
+
+/**
+ * MetadataStoreInterface is able to fetch metadata for a specific workflow.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+interface MetadataStoreInterface
+{
+    public function getWorkflowMetadata(): MetadataBag;
+
+    public function getPlaceMetadata(string $place): ?MetadataBag;
+
+    public function getTransitionMetadata(Transition $transition): ?MetadataBag;
+
+    /**
+     * Returns the metadata for a specific subject.
+     *
+     * This is a proxy method.
+     *
+     * @param null|string|Transition $subject Use null to get workflow metadata
+     *                                        Use a string (the place name) to get place metadata
+     *                                        Use a Transition instance to get transition metadata
+     */
+    public function getMetadata(string $key, $subject = null);
```

Comment:
Okay, I can see how this is useful now. This serves as workaround for not having Ẁorkflow::getMetadata(), Transition::getMetadata() and
Place::getMetadata(). This is useful e.g. when having only list of places retrieved via `definition.places` for retrieving their metadata in same loop, such as
```twig
{% for place in fsm.definition.places %}
    <tr>
        <td>
            {{ place }}
        </td>
        <td>
            {{ workflow_metadata('description', place) }}
        </td>
    </tr>
{% endfor %}
```

instead of doing
```twig
{% for place in fsm.definition.places %}
    <tr>
        <td>
            {{ place }}
        </td>
        <td>
            {{ fsm.definition.placeMetadata(place).get('description') }}
        </td>
    </tr>
{% endfor %}
```

But this functionality can be moved to just twig extension IMHO

Reply:
It's much much easier to use. More over is was asked in the initial issue.
I will keep it fore sure.

Reply:
Fine by me, I can see it's useful now